### ISSUE TITLE
Heartbeat mechanism with Smallrye AMQP connector 

### ISSUE THREAD

**Issue by mnguetsa** [quarkusio/quarkus#39329]
### Describe the bug

Hi everyone,

this is mostly a question as I couldn't change the label. 
Asked the same [here](https://github.com/quarkusio/quarkus/discussions/39246) but did not receive answers so far. 

I am using smallrye reactive messaging with Quarkus to connect to an amqp message broker.
However after a certain idle timeout my application loses connection to the broker due to inactivity and has to reconnect.
I would like to keep the connection without reconnection.
I tried to customize the underlying amqp client by setting TCP KeepAlive to true but it does help with connection losses.
https://quarkus.io/guides/amqp-reference#customizing-the-underlying-amqp-client

My assumption here is that the idle timeout is set at the broker side because I don't think the smallrye amqp has an idle timeout right?

Ideally there should be a mechanism for the client to keep the connection like a heartbeat mechanism in case the broker has a timeout. I could not find something like that in the doc for Smallrye AMQP.
Therefore I need your help in case you know how to solve this.
Thanks

### Output of `java -version`

21

### Quarkus version or git rev

3.8.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

gradle 8.5

**Comment by mnguetsa** [comment]
@geoand @cescoffier Any idea how I could do this? 😬

**Comment by geoand** [comment]
cc @oza

**Comment by cescoffier** [comment]
No, except modifying the AMQP client options, I do not see another approach. 

I would disable the health check and implement a custom one instead.

**Comment by mnguetsa** [comment]
> No, except modifying the AMQP client options, I do not see another approach.
> 
> I would disable the health check and implement a custom one instead.

Hi @cescoffier 
I tried modifying the clients options too. But I realized setTcpKeepAlive(true) had no effect on keeping the connection. Or is there another option I missed in the AMQP client options? So far this is what I did with the AMQP client Options:

```java
@ApplicationScoped
public class AmqpConfigurationProvider {
  @ConfigProperty(name = "amqp-port")
  int port;

  @ConfigProperty(name = "amqp-use-ssl")
  boolean useSsl;

  @Produces
  @Identifier("custom-amqp-client-options")
  public AmqpClientOptions getNamedOptions() {
    return new AmqpClientOptions().setTcpKeepAlive(true).setPort(port).setSsl(useSsl);
  }
}
```

I would use the health check deactivation as last solution. If you know another AMQP option I can try please let me know. 
Thanks a lot

**Comment by cescoffier** [comment]
Which broker are you using? Do they use an idle timeout? Could you potentially send fake frames regularly to keep it open?

**Comment by ozangunalp** [comment]
Probably, you are sending the TCP keep-alive but the server doesn't respect it and closes the connection anyway.

**Comment by cescoffier** [comment]
Cloud-based broker tends to cut connections quickly if there are no L7 frames.
